src/results/mod.rs

```diff
@@ -170,12 +170,15 @@ string_enum!(pub enum FailureReason {
     Unknown => "unknown",
     OOM => "oom",
     Timeout => "timeout",
+    ICE => "ice",
 });
 
+impl Fail for FailureReason {}
```

Comment:
This is a bit unfortunate, but AFAICT it's the nicest way to propagate this up along with the other error metadata into analyze_err

Reply:
There is the `OverrideResult` context:

```rust
if did_ice {
    return Err(e.context(OverrideResult(TestResult::BuildFailure(FailureReason::ICE).into());
} else {
    return Err(e);
}
```

Reply:
I think this feels worse, we don't really want to override the result, just provide the additional metadata that we had an ICE.

Reply:
Hmm, ok, your approach makes sense.